src/Components/ShareDialog.jsx

```diff
@@ -0,0 +1,250 @@
+import React, {useState} from 'react'
+import Paper from '@mui/material/Paper'
+import {makeStyles} from '@mui/styles'
+import ShareIcon from '../assets/3D/Share.svg'
+import ShareClear from '../assets/3D/ShareClear.svg'
+import CheckOn from '../assets/3D/CheckOn.svg'
+import CheckOff from '../assets/3D/CheckOff.svg'
+import Copy from '../assets/3D/Copy.svg'
+import Copied from '../assets/3D/Copied.svg'
+import Checkbox from '@mui/material/Checkbox'
+import TextField from '@mui/material/TextField'
+import {HASH_PREFIX} from './CameraControl'
+import {addHashParams} from '../utils/location'
+import {roundCoord} from '../utils/math'
+
+
+/**
+ * Button to toggle About panel on and off
+ * @param {Number} offsetTop offset tree element
+ * @return {Object} The AboutControl react component.
+ */
+export default function ShareDialogControl({offsetTop, viewer}) {
+  const [open, setOpen] = React.useState(false)
+  const classes = useStyles()
+  return (
+    <div >
+      <div
+        className={classes.about}
+        role='none'
+        onClick={() => {
+          setOpen(!open)
+        }}>
+        <ShareIcon className={classes.icon} />
+      </div>
+      {open &&
+        <ShareDialog
+          viewer={viewer}
+          togglePanel={() => {
+            setOpen(!open)
+          }}
+          offsetTopCssStr={offsetTop} />
+      }
+    </div>)
+}
+
+
+/**
+ * About Panel component
+ * @param {boolean} togglePanel Reactive toggle state for panel.
+ * @param {string} offsetTopCssStr
+ * @return {Component} The AboutPanel react component.
+ */
+function ShareDialog({togglePanel, offsetTopCssStr, viewer}) {
+  const classes = useStyles({offsetTop: offsetTopCssStr})
```

Comment:
Better yet, how about passing down the offsetTop as a number all the way and then turning it into a string below when it's used.